docs/multi-stage-query/reference.md

```diff
@@ -88,93 +118,260 @@ FROM TABLE(
 
-#### Delimited Text Format
+* `uris` - A list of fully-qualified object URIs.
+* `prefixes` - A list of fully-qualified "folder" prefixes.
+* `bucket` and `paths` - A list of objects relative to the given bucket path.
+
+The `S3` function also accepts the following security properties:
 
-Use the `tsv` format to read from an arbitrary delimited (CSV-like) file such as tab-delimited,
-pipe-delimited, etc. This choice selects the Druid `DelimitedInputFormat` class.
+* `accessKeyId` (`VARCHAR`)
+* `secretAccessKey` (`VARCHAR`)
+* `assumeRoleArn` (`VARCHAR`)
 
-| Name | Description | JSON equivalent | Required |
-| ---- | ----------- | --------------- | -------- |
-| `delimiter` | The delimiter which separates fields. | `delimiter` | Yes |
-| `listDelimiter` | The delimiter to use for fields that represent a list of strings. | `listDelimiter` | No |
-| `skipRows` | The number of rows to skip at the start of the file. Default is 0. | `skipHeaderRows` | No |
+The `S3` table function does not support either the `clientConfig` or `proxyConfig`
+JSON properties.
 
-As noted above, MSQ cannot infer schema using headers. Use `EXTEND` instead.
+If you need the full power of the S3 input source, then consider the use of the `extern`
+function, which accepts the full S3 input source serializd as JSON. Alternatively,
+create a catalog external table that has the full set of properties, leaving just the
+`uris` or `paths` to be provided at query time.
+
+Examples, each of which correspond to an example on the
+[S3 input source](../ingestion/native-batch-input-sources.md#s3-input-source) page.
+The examples omit the format and schema; however you must remember to provide those
+in an actual query.
+
+```sql
+  TABLE(S3(uris => ARRAY['s3://foo/bar/file.json', 's3://bar/foo/file2.json']))
+```
+
+```sql
+  TABLE(S3(prefixes => ARRAY['s3://foo/bar/', 's3://bar/foo/']))
+```
+
+```sql
+  -- Not an exact match for the JSON example: the S3 function allows
+  -- only one bucket.
+  TABLE(S3(bucket => 's3://foo`,
+           paths => ARRAY['bar/file1.json', 'foo/file2.json']))
+```
+
+```sql
+  TABLE(S3(uris => ARRAY['s3://foo/bar/file.json', 's3://bar/foo/file2.json'],
+           accessKeyId => 'KLJ78979SDFdS2',
+           secretAccessKey => 'KLS89s98sKJHKJKJH8721lljkd'))
+```
+
+```sql
+  TABLE(S3(uris => ARRAY['s3://foo/bar/file.json', 's3://bar/foo/file2.json'],
+           accessKeyId => 'KLJ78979SDFdS2',
+           secretAccessKey => 'KLS89s98sKJHKJKJH8721lljkd',
+           assumeRoleArn => 'arn:aws:iam::2981002874992:role/role-s3'))
+```
+
+#### Input Format
+
+Each of the table functions above requires that you specify a format using the `format`
+parameter which accepts a value the same as the format names used for `EXTERN` and described
+for [each input source](../ingestion/native-batch-input-sources.md).
+
+#### CSV Format
+
+The `csv` format selects the [CSV input format](../ingestion/data-formats.md#csv).
+Parameters:
+
+* `listDelimiter` (`VARCHAR`)
```

Comment:
Might be good to provide some examples for these different input format types, especially the ones where the user can provide parameters